web/src/pages/blog/index.astro

```diff
@@ -0,0 +1,15 @@
+---
+import MainLayout from "../../layouts/global.astro"
+import PostCard from "../../components/post.astro"
+
+const allPosts = await Astro.glob("./*.mdx")
+---
+
+<MainLayout title="Blog">
+	<section class="container" aria-label="New Blog Posts">
+		<h1 class="h1">New Blog Posts</h1>
+		<div class="post-container">
+			{allPosts.map((post) => <PostCard frontmatter={post.frontmatter} url={post.url} />)}
```

Comment:
IMO inline this instead of using a component. I think it's clearer and makes more sense when you use something like flex to arrange the entries.